src/Eshopworld.Core/Events/BaseEvent.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using Newtonsoft.Json;
+
+namespace Eshopworld.Core.Events
+{
+    /// <summary>
+    /// Standard interface that every event that will be published & consumed from the event streaming bus will expose
+    /// </summary>
+    public abstract class BaseEvent
+    {
+        [JsonConstructor]
+        protected BaseEvent(
+            string identifier,
+            string originService,
+            string createdBy,
+            DateTimeOffset createdOnUtc)
+        {
+            Identifier = identifier;
+            OriginService = originService;
+            CreatedBy = createdBy;
+            CreatedOnUtc = createdOnUtc;
+        }
+        /// <summary>
+        /// Identifier of the event (aggregate's identifier), used as the message key on the bus
+        /// </summary>
+        public string Identifier { get; private set; }
+        /// <summary>
+        /// The service that is responsible for generating the event
+        /// </summary>
+        public string OriginService { get; private set; }
+        /// <summary>
+        /// User reference that is responsibile for generating the event
+        /// </summary>
+        public string CreatedBy { get; private set; }
+        /// <summary>
+        /// Date & time when the event was emitted
+        /// </summary>
+        public DateTimeOffset CreatedOnUtc { get; private set; }
+        /// <summary>
+        /// Code of the brand for which this event was emitted
+        /// </summary>
+        public string TenantCode { get; set; }
```

Comment:
Why is that in the base event? Not all events are tenant specific.

Reply:
Also, some of the events will use tenant code as the identifier for the model as well. I would let this to be put into message payload, if/when required.

Reply:
They shouldn't use the TenantCode as the identifier, it should be unique. 

It being on the base is something we could move, where do you suggest?

Reply:
I believe tenant code should be part of the payload. What's the reason you want it to be exposed to the header? Also, should we consider partitioning events based on tenant codes instead? 

Why would you not use tenant code as the identifier (for brand model at least)?

Reply:
For brands it's fine to be used as an identifier I think. The tenant code on the "header" (see the other repo on Azure) is optional, but does make the envelopes more consistent between all domains where indeed we might be partitioning the data based on the tenant code for example, we should then be able to add this to the library.